### PR TITLE
Added machine learning efficacy

### PR DESCRIPTION
**Pull request by k15z**

### CODE REVIEW

File: sdmetrics/efficacy/base.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    name = ""
+    target_table = ""
+    target_column = ""
+
+    metric_unit = ""
+    metric_goal = Goal.IGNORE
+    metric_domain = (float("-inf"), float("inf"))
+
+    def fit(self, X, y):
+        """This function implements a fit procedure which trains a binary
```
Comment (kveerama) [inline sdmetrics/efficacy/base.py:18]: This doc string seems to be from adversarial metrics. Needs an update to MLefficacy

File: sdmetrics/efficacy/base.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
+import numpy as np
+from sklearn.model_selection import KFold
+
+from sdmetrics import Goal, Metric
+
+
+class MLEfficacy():
+
+    name = ""
+    target_table = ""
```
Comment (kveerama) [inline sdmetrics/efficacy/base.py:10]: target_table_name, target_column_name ??

File: sdmetrics/efficacy/base.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                "table:%s" % self.target_table,
+                "column:%s" % self.target_column,
+            ]),
+            goal=self.metric_goal,
+            unit="delta_%s" % self.metric_unit,
+            domain=(-domain, domain),
+            description="Diff in score on real when trained on synthetic vs real."
+        )
+
+    def _cross_val_score(self, real_table, synthetic_table):
```
Comment (kveerama) [inline sdmetrics/efficacy/base.py:76]: we should change this to evaluate_score, since this is not a cross validation score in the traditional sense. Also, a doc string to explain the process.

File: sdmetrics/efficacy/base.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                "column:%s" % self.target_column,
+            ]),
+            goal=self.metric_goal,
+            unit=self.metric_unit,
+            domain=self.metric_domain,
+            description="Score on the real test set when trained on synthetic data."
+        )
+
+        # Score on synthetic minus score on real. Evaluated on real.
+        domain = self.metric_domain[1] - self.metric_domain[0]
```
Comment (kveerama) [inline sdmetrics/efficacy/base.py:61]: I think this variable should be named delta_domain since this seems to be the domain of the delta_score, and not the domain of the original metric.

File: sdmetrics/efficacy/base.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+        synthetic_X = synthetic_table.loc[:, synthetic_table.columns != self.target_column].values
+        synthetic_y = synthetic_table[self.target_column].values
+
+        delta_scores = []
+        synthetic_scores = []
+        kf = KFold(n_splits=3, shuffle=True)
+
+        for train_index, test_index in kf.split(real_X, real_y):
+            # Train a model on the real dataset.
```
Comment (kveerama) [inline sdmetrics/efficacy/base.py:88]: Train a model on the real dataset and test on the real dataset.

File: sdmetrics/efficacy/base.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+            self.fit(real_X[train_index], real_y[train_index])
+            real_score = self.score(real_X[test_index], real_y[test_index])
+
+            # Train a model on the synthetic dataset.
```
Comment (kveerama) [inline sdmetrics/efficacy/base.py:92]: Train a model on the synthetic dataset and test it on the real test dataset.
